internal/venice-common/src/main/java/com/linkedin/venice/pushmonitor/PushStatusDecider.java

```diff
@@ -272,34 +282,27 @@ protected ExecutionStatus getPartitionStatus(
         if (!currentStatus.equals(COMPLETED)) {
           isLeaderCompleted = false;
         }
-
-        if (currentStatus.equals(ERROR)) {
-          isLeaderInError = true;
+        if (currentStatus.equals(ERROR) && callback != null) {
+          callback.disableReplica(entry.getKey().getNodeId(), partitionStatus.getPartitionId());
         }
-
       }
-
       executionStatusMap.merge(currentStatus, 1, Integer::sum);
     }
 
-    if (executionStatusMap.containsKey(COMPLETED)
-        && executionStatusMap.get(COMPLETED) >= replicationFactor - numberOfToleratedErrors && isLeaderCompleted) {
+    if (executionStatusMap.getOrDefault(COMPLETED, 0) >= (replicationFactor - numberOfToleratedErrors)
```

Comment:
The logic could become more complex since there will be `offline` replicas because of disabling replica.
So there will be two sources of `offline` replicas:
1. The replica hasn't started yet.
2. The replica is disabled.
For the later, if there are 2 or more replicas are disabled, with the current logic, the push will stuck as `started` forever until it gets killed.

I think we will need to maintain a cached version of disabled partitions, and this info will be used to decide whether the `offline` replica is a disabled replica or not-started replica.

When deciding whether a certain partition is errored or not, disabled-offline replica should be treated as `error` replica since Helix won't rebalance `disabled-offline` replica.

Reply:
Another way is to ask helix team to understand whether it is possible to use a different state instead of `offline` for a `disabled` replica.

Reply:
As talked offline, we are going to let the pushjob stuck on 2 error replicas so that restart/rollback of those hosts could try to finish the stuck job.